native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/Engine.java

```diff
@@ -101,74 +101,88 @@ public static Schema getDecodedSchema(BString schemaBString) {
         }
     }
 
-    public static Object executeSubscriptionResource(Environment env, BObject context, BObject service,
-                                                     BObject fieldObject) {
-        Future subscriptionFutureResult = env.markAsync();
-        ExecutionCallback executionCallback = new ExecutionCallback(subscriptionFutureResult);
+    public static Object executeSubscriptionResource(Environment environment, BObject context, BObject service,
+                                                     BObject fieldObject, BObject responseGenerator,
+                                                     Boolean validation) {
         BString fieldName = fieldObject.getObjectValue(INTERNAL_NODE).getStringValue(NAME_FIELD);
         ServiceType serviceType = (ServiceType) TypeUtils.getType(service);
         UnionType typeUnion = TypeCreator.createUnionType(PredefinedTypes.TYPE_STREAM, PredefinedTypes.TYPE_ERROR);
         for (ResourceMethodType resourceMethod : serviceType.getResourceMethods()) {
             if (SUBSCRIBE_ACCESSOR.equals(resourceMethod.getAccessor()) &&
                     fieldName.getValue().equals(resourceMethod.getResourcePath()[0])) {
-                ArgumentHandler argumentHandler = new ArgumentHandler(resourceMethod, context, fieldObject);
-                Object[] args = argumentHandler.getArguments();
-                ObjectType objectType = (ObjectType) TypeUtils.getReferredType(TypeUtils.getType(service));
-                if (objectType.isIsolated() && objectType.isIsolated(resourceMethod.getName())) {
-                    env.getRuntime()
-                            .invokeMethodAsyncConcurrently(service, resourceMethod.getName(), null,
-                                                           null, executionCallback, null, typeUnion, args);
-                } else {
-                    env.getRuntime()
-                            .invokeMethodAsyncSequentially(service, resourceMethod.getName(), null,
-                                                           null, executionCallback, null, typeUnion, args);
+                ArgumentHandler argumentHandler =
+                        new ArgumentHandler(resourceMethod, context, fieldObject, responseGenerator, validation);
+                argumentHandler.validateInputConstraint(environment);
+                if (!argumentHandler.hasConstraintErrors()) {
```

Comment:
This check seems unsafe. After calling the `validateInputConstraint`, checking the `hasConstraintErrors` can be missed. The better way to handle this is to throw an exception there and catch it. Then the compiler itself handles the scenario.

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-graphql/pull/1422/commits/e690ce44c5a3addeedce8feebb473c9785eba1f3